pkg/monitoring/prometheusagent/configmap.go

```diff
@@ -92,18 +93,48 @@ func getServicePriority(cluster *clusterv1.Cluster) string {
 }
 
 // We want to compute the number of shards based on the number of series.
-func getShardsCountForCluster(ctx context.Context, cluster *clusterv1.Cluster, currentShardCount int) (int, error) {
+func (pas PrometheusAgentService) getShardsCountForCluster(
+	ctx context.Context, cluster *clusterv1.Cluster, currentShardCount int) (int, error) {
+
+	clusterShardingStrategy, err := getClusterShardingStrategy(cluster)
+	if err != nil {
+		return 0, errors.WithStack(err)
+	}
+
+	shardingStrategy := sharding.ShardingStrategy{
+		ScaleUpSeriesCount:  pas.MonitoringConfig.ShardingScaleUpSeriesCount,
+		ScaleDownPercentage: pas.MonitoringConfig.ShardingScaleDownPercentage,
+	}.Merge(clusterShardingStrategy)
```

Comment:
```suggestion
	shardingStrategy := pas.MonitoringConfig
	shardingStrategy.Merge(clusterShardingStrategy)
```

Reply:
MonitoringConfig is a struct. Merge will not work

Reply:
We could however create a shardingstrategy in the config instead